### ISSUE TITLE
bin not found causes "ignored" not an error

### ISSUE THREAD

**Issue by sourcefrog** [assert-rs/trycmd#105]
In trying to set up trycmd I had this md file:

    ```trycmd
    $ cargo mutants --help
    ```

This gives a result of `ignored`.

With `--nocapture` I could find out:

```
running 1 test
[              trycmd::runner] 	Substitutions { vars: {"[CWD]": "/home/mbp/src/mutants"}, unused: {"[EXE]"} }
[              trycmd::runner] 	bin="cargo" not found
[              trycmd::runner] 	Case: Ok(
    Output {
        path: "trycmd/hello.trycmd",
        id: Some(
            "4",
        ),
        spawn: Spawn {
            exit: None,
            status: Skipped,
        },
        stdout: None,
        stderr: None,
        fs: Filesystem {
            context: [],
        },
    },
)
Testing trycmd/hello.trycmd:4 ... ignored
test trycmd_tests ... ok
```

It seems to me that "file not found" should probably be a test failure (by default?) not ignored?

**Comment by epage** [comment]
The idea was to gracefully allow mixing untestable examples with testable examples.

We do have the `ignore` tag and it is unlikely people will have testable and untestable examples within the same code block, so we could probably change this.

**Comment by sourcefrog** [comment]
Yep, having an option to ignore seems fine, but maybe it should be the
wrong default....

Martin

**Comment by kleinesfilmroellchen** [comment]
It seems like I had the exact error, just that I couldn't even access detailed error information via `--nocapture`. This is definitely confusing as I first believed it to be a platform-specific error.

**Comment by epage** [comment]
0.14.0 is released with this changed.

**Comment by epage** [comment]
I had forgotten: this is important for when examples are conditionally present: https://github.com/clap-rs/clap/pull/4439

**Comment by kleinesfilmroellchen** [comment]
Interesting; is there a better solution that covers both use cases?

**Comment by AucaCoyan** [comment]
Hi! I'm a bit new to the project. What do you thing about returning an error if a binary is either skipped or doesn't have a name?
The `ignored` keyword tells my common sense that I wrote a `#[skip]` on purpose or I forgot to include in a glob path like `tests/**/*.toml`. 
Having a console output "test is ignored" but not telling me why was a hard thing to debug in my case

**Comment by epage** [comment]
I'd recommend reviewing the  thread for what has been previously considered.

**Comment by veeso** [comment]
I've renamed the function/params to fail_on_unknown_bin

**Comment by epage** [comment]
@veeso what are your thoughts on the alternatives?
